core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -100,10 +103,9 @@ public void start() throws Exception {
       }
       if (gainPrimacy()) {
         mLeaderSelector.waitForState(State.SECONDARY);
-        if (!mRunning) {
-          break;
+        if (ServerConfiguration.getBoolean(PropertyKey.MASTER_JOURNAL_EXIT_ON_DEMOTION)) {
+          stop();
         }
-        losePrimacy();
```

Comment:
Do we not want to call `losePrimacy()` any more?

Reply:
good catch. Updated the logic